### ISSUE TITLE
[Bug] AIPatternScorer._build_trade_setup computes Stop-Loss only from final candle, underestimating risk on multi-candle patterns

### ISSUE THREAD

**Issue by eminsk** [eminsk/yfinance-ta-patterns#10]
### Description
In `AIPatternScorer._build_trade_setup`, the Stop-Loss is calculated using only the `high` or `low` of the single final candle on which the pattern completed:
```python
stop_loss = max(low - buffer, close * 0.001) if is_bullish else high + buffer
```
For multi-candle patterns consisting of 2 to 5 candles (e.g., `CDL3WHITESOLDIERS`, `CDLMORNINGSTAR`, `CDL3OUTSIDE`, `CDLENGULFING`), the true extreme of the formation is almost always located on candle `i-1` or `i-2`, not on the final candle `i`.

### Impact
- For `3WHITESOLDIERS` (BUY), candle `i` has the highest low of the three soldiers. Placing a stop at `low[i] - 0.2*ATR` puts the stop inside the pattern body, underestimating the stop distance by 15–90 pips.
- As a result, the calculated risk is artificially compressed, which distorts the Stop-Loss, Take-Profit targets (`1.5R` and `3.0R`), and the resulting Risk/Reward ratio.

### Proposed Solution
Pass pattern geometry or lookback window $k$ into `_build_trade_setup`:
- $k=1$: Hammer, Shooting Star, Doji, Marubozu, Long Line, etc.
- $k=2$: Engulfing, Harami, Piercing, Dark Cloud Cover, etc.
- $k=3$: Morning Star, Evening Star, 3 White Soldiers, 3 Black Crows, 3 Inside, etc.
- $k=5$: 3 Line Strike, Breakaway, Rise/Fall 3 Methods.

Calculate the stop level based on the pattern's true extreme:
```python
stop_loss = min(lows[i - k + 1 : i + 1]) - buffer  # for BUY
stop_loss = max(highs[i - k + 1 : i + 1]) + buffer # for SELL
```
